### PR TITLE
[WIP] Added program to export point cloud in FBX format

### PR DESCRIPTION
**Pull request by zvrba**

### CODE REVIEW

File: src/software/SfM/main_openMVG2FBX.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    // containing only vertices, w/o any other geometry.
+    //=================================================================================
+
+    {
+      const int point_count = sfm_data.GetLandmarks().size();
+      fbxmesh->InitControlPoints(point_count);
+      FbxVector4* out = fbxmesh->GetControlPoints();
+      for (const auto& lm: sfm_data.GetLandmarks()) {
+        const openMVG::Vec3 &pt = lm.second.X;
+        *out++ = FbxVector4(pt[0], pt[1], pt[2]);
```
Comment (fabiencastan) [inline src/software/SfM/main_openMVG2FBX.cpp]: Could be nice to also export the color available on the landmark: [rgb param declared here](https://github.com/poparteu/openMVG/blob/popart_develop/src/openMVG/sfm/sfm_landmark.hpp#L71)

Comment (zvrba) [inline src/software/SfM/main_openMVG2FBX.cpp]: It's easy, so I'll do it :) A question though: are rgb components floats normalized to [0,1]?

**Comment by fabiencastan** [comment]
Hi,

1. I achieved to convert a scene into FBX, but when I load the result in maya I get an empty node "sfm_node" and no "sfm_mesh".

2. Is it possible to also export the cameras?

**Comment by zvrba** [comment]
I have set up a NIX-based (http://nixos.org) build at home to fix dependency hell once and for all; that's why it took me so long time to answer.

1. I tested this with Blender again; here's a screenshot of the PC and the object graph in the upper right corner. I think Maya may be getting confused because the mesh contains only vertices and no other geometry (lines or triangles).

2. Yes, but... Getting the camera transformation right will be tricky because a) OpenMVG and FBX coordinate systems are different, and b) FBX allows specifying the transformation as TRS (i.e., you cannot give an explicit matrix).  

screenshot from 2016-07-18 08-47-03

**Comment by fabiencastan** [comment]
Alembic files are correctly supported in nuke (even per point color of the PC).

**Comment by zvrba** [comment]
Neither Lightwave, Modo, nor Blender support Alembic files generated by openMVG. I believe that Lightwave doesn't support Alembic at all (Blender definitely doesn't support it, though there's some WIP), while Modo crashes when trying to import the point cloud.

Sure, it'd be nice to have a single format, but Alembic seems to be a niche format which is not well-supported by less-popular 3D packages.

**Comment by fabiencastan** [comment]
Ok.
Yes, there is no file format that is supported by all softwares. So I have no problem with supporting multiple file formats.
I saw in your FBX code that you mention Nuke, so I just precise that Alembic files generated by openMVG are correctly supported in Nuke.

**Comment by zvrba** [comment]
Yes, thanks for the tip about Nuke. Another thing is that we don't want to burden the users with keeping the same data in multiple formats and choosing the right format according to the program which is going to read it.

**Comment by fabiencastan** [comment]
Yes, I completely agree. We need a single file format which works for a complete workflow.

**Comment by fabiencastan** [comment]
@zvrba: What is the status of this PR?

**Comment by zvrba** [comment]
We have a user who needs this feature. I'll pick up this thread some time next week.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -450,6 +451,26 @@ IF(OpenMVG_USE_ALEMBIC)
 ENDIF(OpenMVG_USE_ALEMBIC)
 
 # ==============================================================================
+# FBX
+# ==============================================================================
+# - optional, builds SfM->FBX point cloud converter
+# ==============================================================================
+IF(OpenMVG_FBX_DIR)
```
Comment (simogasp) [inline src/CMakeLists.txt]: it would be better to write a proper FindFBX or use one of those on githu bhttps://goo.gl/3jHIga

Comment (zvrba) [inline src/CMakeLists.txt]: I agree. I suggest that we postpone merging this PR indefinitely, until we have a use for FBX format. After talking directly with NRK, It turned out that they can use Alembic and that they're much more flexible than we thought.

**Pull request closed without merging**